homeassistant/components/influxdb/__init__.py

```diff
@@ -182,6 +191,7 @@ def _generate_event_to_json(conf: Dict) -> Callable[[Dict], str]:
     tags_attributes = conf.get(CONF_TAGS_ATTRIBUTES)
     default_measurement = conf.get(CONF_DEFAULT_MEASUREMENT)
     override_measurement = conf.get(CONF_OVERRIDE_MEASUREMENT)
+    global_ignore_attributes = set(conf.get(CONF_IGNORE_ATTRIBUTES))
```

Comment:
Please use `dict[key]` when we know the key is in the dict.

Reply:
Actually as I'm thinking through your review notes, I'm realizing here that I think need to make a different adjustment. The way the `entity_values` helper class works is it first checks if there are any customizations for the domain of the entity and gets those. Then it checks if there are any customizations for glob and merges those with any domain ones it found. Then finally it looks for entity ID and merges those in too.

Here's the problem, imagine the config was this:
```yaml
influxdb:
  ...
  component_config_domain:
    sensor:
      ignore_attributes:
      - test
  component_config:
    sensor.entity_with_test_attribute
      override_measurement: 'units'
```
In this config I think the user would expect that `sensor.entity_with_test_attribute` would have its measurement overridden to be `units` and its `test` attribute would be ignored. But as the current code is written that isn't what will happen.  Because I am defaulting `ignore_attributes` to `[]` in the config validation, the complete config here is really this:
```yaml
influxdb:
  ...
  component_config_domain:
    sensor:
      ignore_attributes:
      - test
  component_config:
    sensor.entity_with_test_attribute
      ignore_attributes: []
      override_measurement: 'units'
```
Since `entity_values` updates the customizations in order of domain, glob then ID this means the `ignore_attributes` field will be wiped out with an empty array. The user likely would not have expected this to happen.

So I think instead what I need to do is update change [L148](https://github.com/home-assistant/core/blob/0d54dc340957a2101977edb42680c1ceb04fb363/homeassistant/components/influxdb/__init__.py#L148) to this:
```diff
        vol.Optional(CONF_IGNORE_ATTRIBUTES, default=[]): vol.All(
-
        vol.Optional(CONF_IGNORE_ATTRIBUTES): vol.All(
```
and change [L194](https://github.com/home-assistant/core/blob/0d54dc340957a2101977edb42680c1ceb04fb363/homeassistant/components/influxdb/__init__.py#L194) to this:
```diff
    global_ignore_attributes = set(conf.get(CONF_IGNORE_ATTRIBUTES))
-
    global_ignore_attributes = set(conf.get(CONF_IGNORE_ATTRIBUTES, []))
```
What do you think?

Reply:
Ok. I suggest writing a test for this case and first and then fix it.

Reply:
@MartinHjelmare I realized L194 was for the global config, not the entity customization config. The global config can be safely defaulted as part of entity validation and therefore should be changed to `global_ignore_attributes = set(conf[CONF_IGNORE_ATTRIBUTES])`. 

Made that adjustment as well as the new test case and adjustment tin L148, overlapping entity customization works correctly now.